### ISSUE TITLE
Setting size of plot points 'breaks' the plot

### ISSUE THREAD

**Issue by speedreader** [ropensci/plotly#404]
When I manually set the size of the points in a plot (plotly, base graphics in R) the result is a monochrome rectangle. 

First screengrab is of the plot with the default size of points. I'm trying to increase the size of the points, so I set it manually using 'size' (here I used 'size = 20'). That results in the second plot: A rectangular mass of blue. 

I can map the size to a variable in the data. That results in the third screen, so it's not completely broken. However, that's not what I want. I'm trying to make them all the same size only larger. Which, again, looking at the second plot, doesn't seem to be working properly. 
![screen shot 2016-01-16 at 9 52 08 pm](https://cloud.githubusercontent.com/assets/16739809/12375871/6fdf4ef4-bc9c-11e5-9cb7-59fc9095d360.png)
![screen shot 2016-01-16 at 9 52 28 pm](https://cloud.githubusercontent.com/assets/16739809/12375870/6fdecb96-bc9c-11e5-81f2-42f6fee32477.png)
![screen shot 2016-01-16 at 9 52 48 pm](https://cloud.githubusercontent.com/assets/16739809/12375872/6fe193f8-bc9c-11e5-9323-455e6af8aa04.png)

**Comment by cpsievert** [comment]
Please read the contributing guidelines. Issues must be accompanied with reproducible examples -- https://github.com/ropensci/plotly/blob/master/CONTRIBUTING.md#opening-issues

**Comment by speedreader** [comment]
Add 'size'. Plot breaks. Reproducible R code:

xSimpleExample <- rnorm(100, 0, 100)
ySimpleExample <- runif(100, 0, 100)

#This works:
plot_ly(x = xSimpleExample, y = ySimpleExample, mode="markers") %>% layout(plot_bgcolor = "forestgreen")

#This looks like the attached all-blue image:
plot_ly(x = xSimpleExample, y = ySimpleExample, mode="markers", size=5) %>% layout(plot_bgcolor = "forestgreen")

**Comment by pstoyanov** [comment]
Try this, works for me:
`plot_ly(x = xSimpleExample, y = ySimpleExample, mode="markers", marker = list(size = 20))`

Or to vary both size & color of the points, use e.g.
`plot_ly(x = xSimpleExample, y = ySimpleExample, mode="markers", marker = list(size = 20, color = sample(x = c("red", "green", "blue"), size = 100, replace = TRUE)))`

**Comment by speedreader** [comment]
Thanks. I guess I was working incorrectly from one of the examples on the site. So if I put the above in a data.frame instead, as well as add another column value (z), like this:

    exampleDf <- data.frame('x' =rnorm(100, 0, 100), 'y' = rnorm(100, 0, 100), 'z' = rep(1:10,10))

Then this works:

        plot_ly(exampleDf, x = x, y = y, mode="markers", size=z) %>% layout(plot_bgcolor = "forestgreen")

I just assumed that I could assign the size manually. In other words I could just swap out 'z' for an actual number. Apparently, that's not the case as the below again produced a massive blue rectangle (I was originally working with a data.frame when I came across the problem even if my sample above didn't reflect that). 

    plot_ly(exampleDf, x = x, y = y, mode="markers", size=5) %>% layout(plot_bgcolor = "forestgreen")

**Comment by pstoyanov** [comment]
Well, the help file says that `size` is "A variable name or numeric vector to encode the size of markers.", i.e. no mention of a single integer that is recycled for all points. So I guess in hindsight it is obvious, even if not intuitive (for me at least)  :)

The `marker = list(...)` above is the official way to do it. See what you can put in the list here: https://plot.ly/r/reference/#scatter-marker

Generating an extra constant column (e.g. `'z' = 5`) in your data frame and again using `size = z` <b>does not work</b>. It produces the same size markers, i.e. they will look the same for `z = 5` and `z = 20` (even though the correct value of `z` is shown in the hover text).

**Comment by speedreader** [comment]
Strange, thought I posted a reply. Let me try again. The data frame and associated code from my comment above produces the following: 

![screen shot 2016-01-25 at 8 39 33 am](https://cloud.githubusercontent.com/assets/16739809/12558768/091dde76-c357-11e5-8ade-62978564ae9f.png)

So using 'size = z' does produce differently sized markers.

**Comment by cpsievert** [comment]
`plot_ly(size = ...)` is a convenient shortcut if you want to map a variable to size (i.e., it automatically scales arbitrary data values to sizes). `plot_ly()` should eventually know to _not_ do any scaling if a single constant value is provided, but for now, you'll have to use `plot_ly(marker = list(size = ...))` to avoid scaling

**Comment by speedreader** [comment]
Thanks. Will do.
